Deploy Develop

georgreen opened the pull request:
## Resolves Deployment

## Description (what problem you're fixing)

  - Deploy new Version
  -

## Fix (what you did to fix it)

  -
  -

## How to test (describe how to test your PR)

- Clone the repository, install docker, run `make test` within the repository.
- Trigger a build on CircleCI.

Commit:
[Bug] Fix infinite loading issue

  - Assign a default role to users without roles
  - Fix initial data crush bug

Commit:
feat(notifications): slack notifications for master and develop builds

- notify success and failure on staging database upgrade
- notify success and failure on staging deployment
- notify success and failure on production deployment

[ Fixes #161539722 ]

mutoromike committed:
ch(migrate-flask_script-to-flask-cli): set up application to run with flask cli

- initial commit
- remove flask script, configure flask_cli

[Initializes #161470738]

mutoromike committed:
ch(migrate flask script to flask cli): fix tests

- fix failing tests
 [Delivers #161470738]

mutoromike committed:
ch(migrate to flask cli): fix tests

- fix failing tests
 [Delivers #161470738]

mutoromike committed:
ch(migrate to flask cli): update requiremtnts

- update requirements.txt file with new depedencies
 [Finishes #161470738]

mutoromike committed:
ch(migrate from flask-script to flask-cli): fix failing tests

- add flask object to one of the tests
 [Delivers #161470738]

mutoromike committed:
ch(switch to flask-cli from flask-script): fix tests

- update docker files with new details
 [Delivers #161470738]

mutoromike committed:
ch(migrate to flask-cli): remove unused dependencies

- remove unused dependencies
- remove commented lines
[Delivers #161470738]

Commit:
feat(upgrade-backup): database backup

- backup the database before running upgrades on the database

[ Fixes #159519407 ]

mutoromike committed:
bg(fix user details): fix user details
- remove hardcoded token attributes
- fix tests
[fix user login]

mutoromike committed:
ch(switch to flask script): update readme files
- update readme files to reflect new changes
[Delivers ##161470738]

Commit:
feat(CD): deploy the design branch

- deploy the design branch to the staging environment

[ Fixes #162497543]

Commit:
bug(env): add FLASK_APP variable

- add FLASK_APP variable

[ Fixes #162728246]

Commit:
chore(badges): add badges

- configure and add codeclimate to readme
- configure and add circle CI to readme
- configure and add coverage badge to readme

[ Fixes #162779200 ]

Commit:
chore(codacy): add codacy badge

- add the codacy badge to the andela socieites repo

[ Fixes #162779200 ]

mutoromike committed:
configure slack notification helper

mutoromike committed:
configure db queries

mutoromike committed:
add send notification functionality

mutoromike committed:
update requirements file

mutoromike committed:
write tests

mutoromike committed:
fix failing tests

mutoromike committed:
fix failing tests

mutoromike committed:
implement feedback

mutoromike committed:
implement feedback

mutoromike committed:
implement feedback

mutoromike committed:
add success ops notification

mutoromike reviewed (changes requested):
Kindly fix the conflicts, this is needed for the notification feature.

mutoromike committed:
refactor and remove comments

mutoromike committed:
edit notification message

mutoromike committed:
update notification message

mutoromike committed:
add functionality to notify fellow

mutoromike committed:
add comments and logic

mutoromike committed:
fix failing tests

mutoromike committed:
add new environment variable

mutoromike committed:
add new environment variable

mutoromike committed:
improve PR quality

Commit:
chore(pipeline): deploy develop-v2

- create a pipeline for deploying develop-v2

[ Fixes #164068943]

Commit:
chore(upgrade-database): upgrade database for production-v2

- create a pipeline for upgrading the production database

[ Fixes #164100281 ]

Commit:
chore(pipeline-deploy): create a deployment pipeline

- a pipeline to deploy to production

[ Fixes #164100281]

mutoromike committed:
add notification functionality

mutoromike committed:
remove unused imports

mutoromike committed:
add send notification functionality

mutoromike committed:
complete functionality

mutoromike committed:
rebase

mutoromike committed:
add notification functionality

mutoromike committed:
implement fellow notification

mutoromike committed:
refactor and uncomment code

mutoromike committed:
change notification message

frankip committed:
feat(D-level): Implement getting D-level info

- use the user id and the token to retriev data from api
- append the informatio to the data response

[Finishes #164100584]

mutoromike committed:
add send notification functionality

mutoromike committed:
add feature

mutoromike committed:
implement feedback

frankip committed:
ch(loggedActivities): return used and remaining points

- query the society db and return the used and remaining points

[Finishes #164568569]

mutoromike committed:
configure pipenv

mutoromike committed:
implement notify finance feature

mutoromike committed:
fix pr standards

daumie committed:
Update apt sources before running deploy script

daumie committed:
Deploy from current branch for testing purposes

daumie committed:
Update apt sources before upgrading database

daumie committed:
Temporarily add current branch to list of deployable branches

daumie committed:
Remove current test branch from deploying

daumie committed:
use LABEL in-place of MAINTAAAINER tag

daumie committed:
Update app requirements

daumie committed:
Update README.md

frankip committed:
ch(lcohort): Link new cohorts to societies

georgreen commented:
Closing this, no longer relevant.
